media/client/main/include/MediaPipeline.h

```diff
@@ -157,9 +157,11 @@ class MediaPipeline : public IMediaPipelineAndIControlClient, public IMediaPipel
 
     bool renderFrame() override;
 
-    bool setVolume(double volume) override;
+    bool setVolume(double targetVolume, uint32_t duration = 0, EaseType type = EaseType::EASE_LINEAR) override;
```

Comment:
You dont need to include defaults in here, only need it in the interface (one place)